File: pipeline/util/logger.js
/* eslint no-console: 0 */
/*
To log error messages to systemd.
*/

const util = require('util');
const EMERG = 0,
    ALERT = 1,
    CRIT = 2,
    ERR = 3,
    WARNING = 4,
    NOTICE = 5,
    INFO = 6,
    DEBUG = 7;

const prefixes = ['<0>', '<1>', '<2>', '<3>', '<4>', '<5>', '<6>', '<7>'];

function log(level, args) {
    const format = util.format.apply(null, args).replace(/\n/g, `\n${prefixes[level]}`);
    process.stdout.write(`${prefixes[level] + format}\n`);
}

module.exports = {
    emerg: function(...args) {
        log(EMERG, args);
    },
    alert: function(...args) {
        log(ALERT, args);
    },
    crit: function(...args) {
        log(CRIT, args);
    },
    err: function(...args) {
        log(ERR, args);
    },
    warning: function(...args) {
        log(WARNING, args);
    },
    notice: function(...args) {
        log(NOTICE, args);
    },
    info: function(...args) {
        log(INFO, args);
    },
    debug: function(...args) {
        log(DEBUG, args);
    },
};

File: pipeline/util/logger.go
package util

import (
	"fmt"
	"strings"
)

//Logger For Systemd logs
var prefixes = []string{"<0>", "<1>", "<2>", "<3>", "<4>", "<5>", "<6>", "<7>"}

// Debug levels corresponding to the syslog levels
const (
	EMERG int = iota
	ALERT
	CRIT
	ERR
	WARNING
	NOTICE
	INFO
	DEBUG
)

// func log(level DebugLevel, args ...string) {
// 	fmt.Println(prefixes[level], args)
// }

type logger struct{}

// Log is the namespace for the logger functions
var Log = logger{}

func (l logger) Log(level int, str string, args ...interface{}) {
	fmt.Println(prefixes[level] +
		strings.Replace(fmt.Sprintf(str, args...),
			"\n", "\n"+prefixes[level], -1))
}

func (l logger) Emerg(str string, args ...interface{}) {
	l.Log(EMERG, str, args...)
}

func (l logger) Alert(str string, args ...interface{}) {
	l.Log(ALERT, str, args...)
}

func (l logger) Crit(str string, args ...interface{}) {
	l.Log(CRIT, str, args...)
}

func (l logger) Err(str string, args ...interface{}) {
	l.Log(ERR, str, args...)
}

func (l logger) Warning(str string, args ...interface{}) {
	l.Log(WARNING, str, args...)
}

func (l logger) Notice(str string, args ...interface{}) {
	l.Log(NOTICE, str, args...)
}

func (l logger) Info(str string, args ...interface{}) {
	l.Log(INFO, str, args...)
}

func (l logger) Debug(str string, args ...interface{}) {
	l.Log(DEBUG, str, args...)
}
